File: recommendation-service/src/main/java/com/stackroute/domain/BookAppointment.java
package com.stackroute.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BookAppointment {

    @Id
    private String id;
    private Doctor doctor;
    private Patient patient;
    private Date appointmentDate;
    private Integer appointmentId;
    private String appointmentTime;
    private String slot;
    @Transient
    private String key;

}

File: findmeclinic/src/app/bookappointment.ts
import { Patient } from './Patient';
import { Doctor } from './doctor';

export class BookAppointment {

    id: string;
    patient: Patient;
    doctor: Doctor;
    appointmentDate: Date;
    slot: string;
    appointmentId: number;
    key: string;
    appointmentTime: string;
}